packages/upload-client/src/blob.js

```diff
@@ -303,7 +304,38 @@ export async function add(
     })
   }
 
-  return multihash
+  // Ensure the blob has been accepted
+  const acceptReceipt = await new ReceiptPoller(options).poll(
+    nextTasks.accept.task.link()
+  )
+
+  const blocks = new Map(
+    [...acceptReceipt.iterateIPLDBlocks()].map((block) => [
+      `${block.cid}`,
+      block,
+    ])
+  )
+  const site = Delegation.view(
+    {
+      root: /** @type {import('@ucanto/interface').UCANLink} */ (
+        // @ts-ignore Property
+        acceptReceipt.out.ok.site
```

Comment:
You should check for `error` and then you don't need to `ts-ignore` here.

Reply:
I don't think I need at all anymore, as previously the receipt acquisition method could return null. Now it's either a receipt or it throws.